insights/client/connection.py

```diff
@@ -299,167 +376,152 @@ def get_proxies(self):
         self.proxies = proxies
         self.proxy_auth = proxy_auth
 
+    # -LEGACY-
     def _legacy_test_urls(self, url, method):
         """
-        Actually test the url
+        Send a request to test a URL (legacy version)
+
+        Parameters
+            url     - URL to send a test request to
+            method  - HTTP method to use
+
+        Returns
+            True on success
+            False on failure
         """
         # tell the api we're just testing the URL
         test_flag = {'test': 'test'}
         url = urlparse(url)
-        test_url = url.scheme + "://" + url.netloc
-        last_ex = None
-        paths = (url.path + '/', '', '/r', '/r/insights')
-        for ext in paths:
-            try:
-                logger.log(NETWORK, "Testing: %s", test_url + ext)
-                if method is "POST":
-                    test_req = self.session.post(
-                        test_url + ext, timeout=self.config.http_timeout, data=test_flag)
-                elif method is "GET":
-                    test_req = self.session.get(test_url + ext, timeout=self.config.http_timeout)
-                logger.log(NETWORK, "HTTP Status Code: %d", test_req.status_code)
-                logger.log(NETWORK, "HTTP Status Text: %s", test_req.reason)
-                logger.log(NETWORK, "HTTP Response Text: %s", test_req.text)
-                # Strata returns 405 on a GET sometimes, this isn't a big deal
-                if test_req.status_code in (200, 201):
-                    logger.info(
-                        "Successfully connected to: %s", test_url + ext)
-                    return True
-                else:
-                    logger.info("Connection failed")
-                    return False
-            except requests.ConnectionError as exc:
-                last_ex = exc
-                logger.error(
-                    "Could not successfully connect to: %s", test_url + ext)
-                print(exc)
-        if last_ex:
-            raise last_ex
+        test_url = url.scheme + "://" + url.netloc + url.path
+        logger.log(NETWORK, "Testing: %s", test_url)
+        if method is "POST":
+            test_req = self.post(test_url, data=test_flag)
```

Comment:
No Content-Type header? Or maybe rather `json=test_flag`?

```suggestion
            test_req = self.post(test_url, json=test_flag)
```